cli/src/template_builder.rs

```diff
@@ -813,6 +813,40 @@ fn build_binary_operation<'a, L: TemplateLanguage<'a> + ?Sized>(
             };
             Ok(L::Property::wrap_property(out))
         }
+        BinaryOp::Add | BinaryOp::Sub | BinaryOp::Mul | BinaryOp::Div | BinaryOp::Rem => {
+            let lhs = expect_integer_expression(language, diagnostics, build_ctx, lhs_node)?;
+            let rhs = expect_integer_expression(language, diagnostics, build_ctx, rhs_node)?;
+            let out = lhs.and_then(move |l| {
```

Comment:
`(lhs, rhs).and_then(|(l, r)|` (no short-circuiting is needed here)

Maybe we can also move `match op` out of the property builder chain. It's a little bit nicer that bad `unreachable!()` condition would be detected at the parsing phase.

Reply:
I applied both changes, although pulling the `match op` out does make it significantly more verbose.

Reply:
Maybe you can add helper function of `(lhs, rhs, f) -> BoxedTemplateProperty<'a, i64>` type? Then, each operation will be like `apply(lhs, rhs, i64::checked_add, "Attempt to add with overflow")`.